lib/enterprise-event-stream.js

```diff
@@ -0,0 +1,252 @@
+// Private
+// ------------------------------------------------------------------------------
+
+const DEFAULT_OPTIONS = Object.freeze({
+	pollingInterval: 60,	// seconds
+	chunkSize: 500
+
+});// ------------------------------------------------------------------------------
+// Public
+// ------------------------------------------------------------------------------
+
+/**
+ * Stream of Box enterprise events.
+ *
+ * By default, the stream starts from the current time.
+ * Pass 'startDate' to start from a specific time.
+ * Pass 'streamPosition' to start from a previous stream position, or '0' for all available past events (~1 year).
+ * Once the stream catches up to the current time, it will begin polling every 'pollingInterval' seconds.
+ * If 'pollingInterval' = 0, then the stream will end when it catches up to the current time (no polling).
+ *
+ * @param {BoxClient} client - The client to use to get events
+ * @param {Object} [options] - Options
+ * @param {string} [options.streamPosition] - The stream position to start from (pass '0' for all past events)
+ * @param {string} [options.startDate] - The date to start from
+ * @param {string} [options.endDate] - The date to end at
+ * @param {string} [options.eventTypeFilter] - Comma-separated list of event types to return
+ * @param {int} [options.pollingInterval=60] - Polling interval (in seconds).  Pass 0 for no polling.
+ * @param {int} [options.chunkSize=500] - Number of events to fetch per call (max = 500)
+ * @constructor
+ * @extends Readable
+ */
+function EnterpriseEventStream(client, options) {
+
+	Readable.call(this, {
+		objectMode: true
+	});
+
+	/**
+	 * @var {BoxClient} - The client for making API calls
+	 * @private
+	 */
+	this._client = client;
+
+	/**
+	 * @var {Object} - Options
+	 * @private
+	 */
+	this._options = Object.assign({}, DEFAULT_OPTIONS, options);
+
+	// Handle the case where the caller passes streamPosition = 0 instead of streamPosition = '0'.
+	if (!this._options.startDate && !this._options.streamPosition && this._options.streamPosition !== 0) {
+		// If neither startDate nor streamPosition is specified, start from the current time.
+		this._options.startDate = new Date().toISOString().replace(/\.000Z$/, '-00:00');
+	}
+
+	/**
+	 * @var {?string} - The current stream position
+	 * @private
+	 */
+	this._streamPosition = this._options.streamPosition;
+}
+
+util.inherits(EnterpriseEventStream, Readable);
+
+/**
+ * @returns {?string} - Returns null if no events have been fetched from Box yet.
+ */
+EnterpriseEventStream.prototype.getStreamPosition = function() {
+	return this._streamPosition;
+};
+
+/**
+ * Read and set the stream position and start date from a file.
+ *
+ * @param {string} file - A filename or file descriptor
+ * @param {Function} callback - Passed the stream position and start date
+ * @returns {void}
+ */
+EnterpriseEventStream.prototype.readStreamPositionFromFile = function(file, callback) {
```

Comment:
This seems like the wrong level for the SDK; users can read and persist state however they want.  Why not expose a `getState()` method that returns the necessary state and a `resume()` method that takes the expected state object?  That way consumers could store state in Redis, etc. if they want